File: deployments/containers/runners/python/execute.py
"""
Default execute function that raises an error.
This file should be replaced by mounting a custom execute.py
or by building a derived image with a custom implementation.
"""

from typing import Any, Dict


def execute_function(input_data: Dict[str, Any]) -> Dict[str, Any]:
    """
    Execute function - must be implemented by user.

    Args:
        input_data: The input data

    Returns:
        The output data

    Raises:
        NotImplementedError: Always, as this is a placeholder
    """
    raise NotImplementedError(
        "No execution function provided. "
        "Either mount a custom execute.py at /app/execute.py "
        "or use a pre-built generator image (e.g., archesai/generator-custom)."
    )

File: deployments/containers/runners/node/src/execute.ts
/**
 * Default execute function that throws an error.
 * This file should be replaced by mounting a custom execute.ts/execute.js
 * or by building a derived image with a custom implementation.
 */

import type { Executor } from "#runner";

/**
 * Execute function - must be implemented by user
 * @param input - The input data
 * @returns The output data
 * @throws Error if no custom implementation is provided
 */
async function executeFunction(_input: unknown): Promise<unknown> {
  throw new Error(
    "No execution function provided. " +
      "Mount a custom execute.ts at /app/execute.ts ",
  );
}

/**
 * Export the execute function as an Executor type
 */
const _executeFunction: Executor<unknown, unknown> = executeFunction;

export { _executeFunction as executeFunction };
